department-of-veterans-affairs/caseflow
# Add Remand Appeal Association to CavcRemand model

Resolves #15710 / [CASEFLOW-688](https://vajira.max.gov/browse/CASEFLOW-688)

### Description
Adds an association to the created Remand appeal for a CavcRemand object

### Acceptance Criteria
  - [x] The remand_appeal_id on a CavcRemand is the appeal id of the created CAVC appeal
  - [x] Relevant tests, factories, and seeds are updated
  - [x] Code in issues below  are updated to reflect new association (see [Technical notes](https://github.com/department-of-veterans-affairs/caseflow/issues/15719#) for suggestions)
    1. [When displaying the information about the remand](https://github.com/department-of-veterans-affairs/caseflow/blob/f6c37b6587eb6bd083ff03c380cd20c8e6a69ec2/app%2Fmodels%2Fappeal.rb#L358) on a cavc appeal, we try to track down a remand associated with the original appeal (will break if there is more than on remand for a single appeal)
    1. [When we return the newly created cavc appeal after processing a remand](https://github.com/department-of-veterans-affairs/caseflow/blob/7b4d8e0f8c8b0524c4ca2592cb0121ecec1644ff/app%2Fcontrollers%2Fcavc_remands_controller.rb#L10), we look for any cavc appeal with that docket number (will break if there is more than on remand for a single appeal)

### Testing Plan
1. Create a dispatched appeal on the console:
    `appeal = FactoryBot.create(:appeal, :dispatched)`
    `FactoryBot.create_list(:request_issue, 2, :rating, :with_rating_decision_issue, decision_review: appeal, veteran_participant_id: Veteran.first.participant_id, contested_issue_description: "foo desc", notes: "bar notes")`
2. Go to that appeal on the UI
3. Create a jmr cavc appeal
4. Confirm creation still succeeds
5. Find the new cavc appeal on the command line
6. Confirm you can still find its cavc_remand:
    `remand_appeal.cavc_remand`

### Database Changes
*Only for Schema Changes*

* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [x] Appropriate indexes added (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] ~Any non-obvious semantics or logic useful for interpreting database data is documented at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)~
